pkg/integrations/install/shims.go

```diff
@@ -0,0 +1,32 @@
+package install
+
+import (
+	v1 "github.com/grafana/agent/pkg/integrations"
+	v2 "github.com/grafana/agent/pkg/integrations/v2"
+	metricsutils "github.com/grafana/agent/pkg/integrations/v2/metricsutils"
+)
+
+// Perform a migration of v1 integrations which do not yet have a v2
+// counterpart. These integrations will be registered as Singletons
+// to maintain existing behavior.
+//
+// To migrate a v1 integration to a v2 migration with support for multiple
+// instances, v1 integrations must manually migrate themselves by calling
+// v2.RegisterDynamic directly.
+func init() {
```

Comment:
I wanted to log that we were doing this, but by the time a logger is created it's already too late and integrations would've been unmarshaled.